### PR TITLE
Remove TensorFlow take_along_axis workarounds

### LINKED ISSUE #1312: Remove workaround for `take_along_axis`
Our `backend` module is now a fairly clean Keras 3 API, but we have one major work around remaining. During our port to Keras 3 we had to hack up `take_along_axis` on the tensorflow backend to support dynamic shape.

With a Keras 3 [local development environment](https://github.com/keras-team/keras-nlp/blob/master/CONTRIBUTING.md#linux-recommended), with the tensorflow backend:
- Delete the [take_along_axis](https://github.com/keras-team/keras-nlp/blob/11bece833200548c181810980ff50546da4d748a/keras_nlp/backend/ops.py#L26C9-L26C24) hack.
- Some tests should begin fail because of dynamic shape issues.
- Either, move to another `ops` usage that avoid this issue, or upstream the work-around into Keras 3 on the tf backend (add tests, make more robust).

If we upstream the change into Keras 3, we may have to keep the compat shim around until we get a new Keras release with our fix. That is ok.

### PR DESCRIPTION
**Pull request by GargiGupta-io**
Tested on the latest TensorFlow and Keras versions — the original dynamic shape issue with `take_along_axis` appears to be fixed. So I removed the TensorFlow-specific workaround and switched everything to use `keras.ops.take_along_axis` uniformly. Cleaner code and more consistent across backends.

Also added tests for dynamic mask positions to catch any regressions.

Fixes #1312.

### CODE REVIEW

**Comment by laxmareddyp** [comment]
Hi @GargiGupta-io Thanks for the PR..! Could you please rebase your branch ?

File: keras_hub/src/models/roformer_v2/roformer_v2_masked_lm_test.py [hunk 1]
```diff
@@ -51,3 +53,31 @@ def test_masked_lm_basics(self):
...
+        )
+        model = RoformerV2MaskedLM(
+            backbone=backbone,
+            preprocessor=None,
+        )
+        inputs = {
+            "token_ids": np.array([[1, 2, 3, 4, 5]] * 2, dtype="int32"),
+            "segment_ids": np.zeros((2, 5), dtype="int32"),
+            "padding_mask": np.ones((2, 5), dtype="int32"),
+        }
```
[the remark this answers is not in the document]
Comment (GargiGupta-io) [reply keras_hub/src/models/roformer_v2/roformer_v2_masked_lm_test.py:75]: Addressed in `47884d3e`: the undeclared `padding_mask` input was removed from the dynamic-mask test.

**Comment by GargiGupta-io** [comment]
Rebased onto the latest `master` in `47884d3e`. All CI checks are now passing.

**Comment by GargiGupta-io** [comment]
Your testing confirms the original dynamic shape issue with take_along_axis has been fixed in recent TensorFlow versions. The tests pass and dynamic mask positions work correctly across all backends, so the workaround can be safely removed.

**Review by laxmareddyp — APPROVED** [review]
Thanks for the PR..! LGTM

**Comment by GargiGupta-io** [comment]
@divyashreepathihalli All checks passing and approved the pr is ready to merge. Please look into it whenever u r free

**Review by divyashreepathihalli — APPROVED** [review]

**Comment by divyashreepathihalli** [comment]
@jules Verify the changes in this PR please.
